components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -219,7 +221,32 @@ func (wsc *WorkspaceController) handleWorkspaceRunning(ctx context.Context, ws *
 	span, ctx := opentracing.StartSpanFromContext(ctx, "handleWorkspaceRunning")
 	defer tracing.FinishSpan(span, &err)
 
-	return ctrl.Result{}, wsc.operations.SetupWorkspace(ctx, ws.Name)
+	var imageInfo *workspacev1.WorkspaceImageInfo = nil
+	if ws.Status.ImageInfo == nil {
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+		id, err := wsc.runtime.WaitForContainer(ctx, ws.Name)
+		if err != nil {
```

Comment:
We should ignore the `Cancel` and `Timeout` errors here, and just jump over it instead of "failing" the operation to keep the impact local.  :+1:

Reply:
Love it! :orange_heart: